pkg/capacity/capacity.go

```diff
@@ -377,24 +389,25 @@ func (c *Controller) onSCDelete(sc *storagev1.StorageClass) {
 	if sc.Provisioner != c.driverName {
 		return
 	}
+	logger := c.logger.WithName("onSCDelete").WithValues("storageclass", klog.KObj(sc))
 
-	klog.V(3).Infof("Capacity Controller: storage class %s was removed", sc.Name)
+	logger.V(3).Info("removed")
```

Comment:
Can you show an example of the output of this changed log msg vs the original log msg?

Reply:
Looks like I have a coverage gap in capacity_test.go - onSCDelete is not called. Will fix.

In the meantime, here's the corresponding output from onSCAddedOrUpdated, as printed by `go test`. With these changes:
```
capacity.go:373: INFO onSCAddOrUpdate: updated or added storageclass="triple-sc"
...
capacity.go:481: INFO onSCAddOrUpdate: enqueuing storageclass="triple-sc" workitem={segment:0x27ef430 storageClassName:triple-sc}
```

Without them:
```
I0608 17:48:40.721547  866697 capacity.go:361] Capacity Controller: storage class triple-sc was updated or added
...
I0608 17:48:40.721526  866697 capacity.go:468] Capacity Controller: enqueuing {segment:0x27e7350 storageClassName:other-sc}
```

Note that the "enqueuing" messages without this PR lacks context. It's not clear why `addWorkItem` was called. With contextual logging, the `onSCAddOrUpdate` function name and the storage class get passed down and are added to the log message.

This is important once things start to happen in parallel. When everything is sequential, one can read the log from top to bottom and remember which values were logged earlier. But when run in parallel, it's not clear whether log output actually follows the one printed directly above it.

This is a problem in our driver logs where it is hard to associate a gRPC error response with the corresponding call.

Reply:
Here's the log output for a complete testcase:
```
=== RUN   TestRefresh
=== RUN   TestRefresh/truncated_topology
=== PAUSE TestRefresh/truncated_topology
=== CONT  TestRefresh/truncated_topology
    capacity.go:338: INFO onTopologyChanges: topology changed added=[0x27ef430 = layer0: foo+ layer1: X+ layer2: A 0x27ef450 = layer0: foo+ layer1: X+ layer2: B] removed=[]
    capacity.go:373: INFO onSCAddOrUpdate: updated or added storageclass="direct-sc"
    capacity.go:481: INFO onSCAddOrUpdate: enqueuing storageclass="direct-sc" workitem={segment:0x27ef430 storageClassName:direct-sc}
    capacity.go:481: INFO onSCAddOrUpdate: enqueuing storageclass="direct-sc" workitem={segment:0x27ef450 storageClassName:direct-sc}
    capacity.go:373: INFO onSCAddOrUpdate: updated or added storageclass="triple-sc"
    capacity.go:481: INFO onSCAddOrUpdate: enqueuing storageclass="triple-sc" workitem={segment:0x27ef430 storageClassName:triple-sc}
    capacity.go:481: INFO onSCAddOrUpdate: enqueuing storageclass="triple-sc" workitem={segment:0x27ef450 storageClassName:triple-sc}
    capacity.go:338: INFO onTopologyChanges: topology changed added=[0x27ef430 = layer0: foo+ layer1: X+ layer2: A 0x27ef450 = layer0: foo+ layer1: X+ layer2: B] removed=[]
    capacity.go:481: INFO onTopologyChanges: enqueuing workitem={segment:0x27ef430 storageClassName:direct-sc}
    capacity.go:481: INFO onTopologyChanges: enqueuing workitem={segment:0x27ef450 storageClassName:direct-sc}
    capacity.go:481: INFO onTopologyChanges: enqueuing workitem={segment:0x27ef430 storageClassName:triple-sc}
    capacity.go:481: INFO onTopologyChanges: enqueuing workitem={segment:0x27ef450 storageClassName:triple-sc}
    capacity.go:276: INFO prepare: initial state topology segments=2 storage classes=2 potential CSIStorageCapacity objects=4
    capacity.go:288: INFO prepare: checking for existing CSIStorageCapacity objects
--- PASS: TestRefresh (0.00s)
```

Note that this would be impossible to do without this PR because log output from different test cases would be mixed.